collect_app/src/main/java/org/odk/collect/android/backgroundwork/CollectBackgroundWorkManager.java

```diff
@@ -0,0 +1,68 @@
+package org.odk.collect.android.backgroundwork;
+
+import androidx.work.WorkInfo;
+import androidx.work.WorkManager;
+
+import com.evernote.android.job.Job;
+import com.evernote.android.job.JobManager;
+import com.google.common.util.concurrent.ListenableFuture;
+
+import org.odk.collect.android.tasks.ServerPollingJob;
+import org.odk.collect.android.upload.AutoSendWorker;
+import org.odk.collect.utilities.BackgroundWorkManager;
+
+import java.util.List;
+import java.util.Set;
+
+import timber.log.Timber;
+
+
+/**
+ * Abstraction that sits on top of {@link com.evernote.android.job.JobManager} and
+ * {@link androidx.work.WorkManager}. Implementation of {@link BackgroundWorkManager} which objects
+ * can use to interact with background work without having to care what underlying framework is
+ * being used.
+ */
+public class CollectBackgroundWorkManager implements BackgroundWorkManager {
+
+    @Override
+    public boolean isRunning(String tag) {
+        switch (tag) {
+            case AutoSendWorker.TAG: {
+                return isWorkManagerWorkRunning(tag);
+            }
+
+            case ServerPollingJob.TAG: {
+                return isJobManagerWorkRunning(tag);
+            }
+
+            default:
+                throw new IllegalArgumentException("CollectBackgroundWorkManager doesn't know about " + tag);
+        }
+    }
+
+    private boolean isJobManagerWorkRunning(String tag) {
```

Comment:
I feel anxious about the test you removed that verified those two methods. Those were android tests because starting evernote job / worker didn't work with Robolectric I don't know why. It required some time but at least it was verified and we were sure it won't cause any problem during migration. Now you moved the code here and it looks fine but generally when you refactor something it's good to add tests in this case we have refactoring and removing tests.

Reply:
I can completely see where you're coming from here. The motivation behind changing the tests is that locally and on CI they were failing on and off as the waits wouldn't always work. 

I looked at faking out `JobManager`/`WorkManager` in the `StorageMigrator` tests but was unable to due to `JobManager` being a `final` class. Generally I'd wrap a depedency like this as I'm not trying to test that JobManager works but that our code that interacts with it does. Granted here we're wrapping a chunk of our own logic so it's not the "thinnest" wrapper.

We could of course test the wrapper (`CollectBackgroundWorkManager`) to give us more confidence. I'm comfortable with it being at the "edge" of our system enough that it doesn't need that but that's a pretty subjective statement. If you're feeling really uncomfortable with it I can have a stap at tests for the `CollectBackgroundWorkManager`. I'm not sure how feasible it would be for `JobManager` (although at worst we could wrap it in a veru light method proxy interface for tests) but WorkManager provides faking tools as far as I've seen.

Reply:
Ok so let's merge it as is not to block other prs but if you have time you can add test if it's difficult please just let the QA team know about it so that they can verify this case manually one more time.

Reply:
Ok that sounds good. I'll have a look at that. Go ahead and merge when you're ready!